src/Microsoft.TemplateEngine.Abstractions/PhysicalFileSystem/IPhysicalFileSystem.cs

```diff
@@ -126,5 +126,14 @@ public interface IPhysicalFileSystem
         /// <param name="file">The file to set last write time for.</param>
         /// <param name="lastWriteTimeUtc">the time to set.</param>
         void SetLastWriteTimeUtc(string file, DateTime lastWriteTimeUtc);
+
+        /// <summary>
+        /// If target is a subpath of relativeTo a relative bath from relativeTo to subpath will be returned.
+        /// </summary>
+        /// <param name="target">Path to be converted to relative if possible.</param>
+        /// <param name="relativeTo">Base of the relative path to be returned.</param>
+        /// <param name="desiredDirectorySeparator">Desired separator of directories in resulting path. If not specified, defaults to <see cref="Path.DirectorySeparatorChar"/>.</param>
+        /// <returns></returns>
+        string PathRelativeTo(string target, string relativeTo, char desiredDirectorySeparator = char.MaxValue);
```

Comment:
an enum might be better here (unix, windows, currentOs) - I'll update

Reply:
Can be also based on executing OS.